### PR TITLE
feat!: move to swift based framework for didcomm, jose and peer did

### PR DESCRIPTION
**Pull request by goncalo-frade-iohk**
BREAKING CHANGE: Updated Pluto and Message public interface

This changes are required to provide the following.
- the old didcomm library didnt allow for extra headers that was essencial to prism mediator
- this peer did library is updated with last specifications specs required by the agent and prism mediator
- this jose library provides a full implementation of the jose capabilities we require

The amount of changes is big because the peer did changes required a few changes on how we process keys and resolve secrets

Besides that since the previous libraries were not build in swift, we are actually getting rid of 37MB of framework size

### My PR contains...
* [ ] No code changes (changes to documentation, CI, metadata, etc.)
* [x] Bug fixes (non-breaking change which fixes an issue)
* [x] Improvements (misc. changes to existing features)
* [ ] Features (non-breaking change which adds functionality)

### My changes...
* [x] are breaking changes
* [ ] are not breaking changes
* [x] If yes to the above: I have updated the documentation accordingly

### Documentation
* [ ] My changes do not require a change to the project documentation
* [x] My changes require a change to the project documentation
* [x] If yes to the above: I have updated the documentation accordingly

### Tests
* [ ] My changes can not or do not need to be tested
* [x] My changes can and should be tested by unit and/or integration tests
* [x] If yes to the above: I have added tests to cover my changes
* [ ] If yes to the above: I have taken care to cover edge cases in my tests

### CODE REVIEW

File: Sample/DIDChat/DIDChat.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -404,7 +404,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_ASSET_PATHS = "\"DIDChat/Preview Content\"";
-				DEVELOPMENT_TEAM = 89TW38X994;
+				DEVELOPMENT_TEAM = "";
```
Comment (elribonazo) [inline Sample/DIDChat/DIDChat.xcodeproj/project.pbxproj:407]: nice one

**Review by elribonazo — APPROVED** [review]

File: AtalaPrismSDK/Castor/Sources/CastorImpl+Public.swift [hunk 2]
```diff
@@ -138,13 +138,14 @@ extension CastorImpl: Castor {
...
-            services: []
-        ).computeEcnumbasis(did: did, publicKey: publicKey)
+//        logger.debug(message: "Getting ecnumbasis", metadata: [
+//            .maskedMetadataByLevel(key: "DID", value: did.string, level: .debug)
+//        ])
+//        return try CreatePeerDIDOperation(
+//            autenticationPublicKey: publicKey,
+//            agreementPublicKey: publicKey,
+//            services: []
+//        ).computeEcnumbasis(did: did, publicKey: publicKey)
```
Comment (cristianIOHK) [inline AtalaPrismSDK/Castor/Sources/CastorImpl+Public.swift:148]: Do we need to keep this commented code?

File: AtalaPrismSDK/Castor/Sources/Operations/CreatePeerDIDOperation.swift [hunk 3]
```diff
@@ -1,172 +1,58 @@
...
             format: .jwk,
-            value: jwkString,
-            type: .jsonWebKey2020
+            key: publicKey.raw,
+            type: .agreement(.jsonWebKey2020)
         )
     }
-
-    private func authenticationFromPublicKey(publicKey: PublicKey) throws -> VerificationMaterialAuthentication {
+//
```
Comment (cristianIOHK) [inline AtalaPrismSDK/Castor/Sources/Operations/CreatePeerDIDOperation.swift:46]: Can we remove this line?

File: Sample/DIDChat/DIDChat/Modules/MediatorView/MediatorViewModel.swift [hunk 4]
```diff
@@ -36,7 +36,10 @@ class MediatorViewModelImpl: MediatorViewModel {
                 await MainActor.run { [weak self] in
                     self?.routeToContactsList = true
                 }
-            } catch {}
+            } catch {
+                print(error)
+                print()
```
Comment (hamada147) [inline Sample/DIDChat/DIDChat/Modules/MediatorView/MediatorViewModel.swift:41]: Remove Print from code.
```suggestion
```

File: AtalaPrismSDK/Castor/Sources/CastorImpl+Public.swift (hunk 2 above)
Comment (goncalo-frade-iohk) [reply AtalaPrismSDK/Castor/Sources/CastorImpl+Public.swift]: Done

File: AtalaPrismSDK/Castor/Sources/Operations/CreatePeerDIDOperation.swift (hunk 3 above)
Comment (goncalo-frade-iohk) [reply AtalaPrismSDK/Castor/Sources/Operations/CreatePeerDIDOperation.swift]: Done

File: Sample/DIDChat/DIDChat/Modules/MediatorView/MediatorViewModel.swift (hunk 4 above)
Comment (goncalo-frade-iohk) [reply Sample/DIDChat/DIDChat/Modules/MediatorView/MediatorViewModel.swift]: One of them I will leave since it is in the demo ;)
